impeller/renderer/renderer_unittests.cc

```diff
@@ -1046,6 +1046,137 @@ TEST_P(RendererTest, VertexBufferBuilder) {
+      vertex_builder.CreateVertexBuffer(*context->GetResourceAllocator());
+  ASSERT_TRUE(vertex_buffer);
+
+  desc->SetSampleCount(SampleCount::kCount4);
+
+  auto bridge = CreateTextureForFixture("bay_bridge.jpg");
+  auto boston = CreateTextureForFixture("boston.jpg");
+  ASSERT_TRUE(bridge && boston);
+  auto sampler = context->GetSamplerLibrary()->GetSampler({});
+  ASSERT_TRUE(sampler);
+
+  static int stencil_reference_write = 0xFF;
+  static int stencil_reference_read = 0x1;
+  std::vector<uint8_t> stencil_contents;
+  static int last_stencil_contents_reference_value = 0;
+  Renderer::RenderCallback callback = [&](RenderTarget& render_target) {
+    auto buffer = context->CreateCommandBuffer();
+    if (!buffer) {
+      return false;
+    }
+    buffer->SetLabel("Playground Command Buffer");
+
+    {
+      // Configure the stencil attachment for the test.
+      RenderTarget::AttachmentConfig stencil_config;
+      stencil_config.load_action = LoadAction::kLoad;
+      stencil_config.store_action = StoreAction::kDontCare;
+      stencil_config.storage_mode = StorageMode::kHostVisible;
+      render_target.SetupStencilAttachment(*context,
+                                           render_target.GetRenderTargetSize(),
+                                           true, "stencil", stencil_config);
+      // Fill the stencil buffer with an checkerboard pattern.
+      const auto target_width = render_target.GetRenderTargetSize().width;
+      const auto target_height = render_target.GetRenderTargetSize().height;
+      const size_t target_size = target_width * target_height;
+      if (stencil_contents.size() != target_size ||
+          last_stencil_contents_reference_value != stencil_reference_write) {
+        stencil_contents.resize(target_size);
+        last_stencil_contents_reference_value = stencil_reference_write;
+        for (int y = 0; y < target_height; y++) {
+          for (int x = 0; x < target_width; x++) {
+            const auto index = y * target_width + x;
+            const auto kCheckSize = 64;
+            const auto value =
+                (((y / kCheckSize) + (x / kCheckSize)) % 2 == 0) *
+                stencil_reference_write;
+            stencil_contents[index] = value;
+          }
+        }
+      }
+      if (!render_target.GetStencilAttachment()->texture->SetContents(
+              stencil_contents.data(), stencil_contents.size(), 0, false)) {
+        VALIDATION_LOG << "Could not upload stencil contents to device memory";
+        return false;
+      }
+      auto pass = buffer->CreateRenderPass(render_target);
+      if (!pass) {
+        return false;
+      }
+      pass->SetLabel("Stencil Buffer");
+      ImGui::Begin("Controls", nullptr, ImGuiWindowFlags_AlwaysAutoResize);
+      ImGui::SliderInt("Stencil Write Value", &stencil_reference_write, 0,
+                       0xFF);
+      ImGui::SliderInt("Stencil Compare Value", &stencil_reference_read, 0,
+                       0xFF);
```

Comment:
Nit: It would be nice if the front face stencil behavior could be checked separately from this tool. One way would be to:
1. set the `stencil_compare` to `kLessEqual`, and then
2. add an `x_mirror` bool to the controls that, when multiplies the mvp by `Matrix::MakeScale(Vector2(-1, -1))`

Reply:
Done.

Reply:
Whoops, means to say invert just x with `Matrix::MakeScale(Vector2(-1, 1))` and then the image would flip + the checkerboard pattern should invert.

Reply:
And keep the front face as `kGreater`, but just change the back face to `kLessEqual`.